frontend/src/app/data-export/data-export.component-alt.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2014-2021 Bjoern Kimminich.
+ * SPDX-License-Identifier: MIT
+ */
+
+import { Component, OnInit } from '@angular/core'
+import { FormControl, Validators } from '@angular/forms'
+import { ImageCaptchaService } from '../Services/image-captcha.service'
+import { DataSubjectService } from '../Services/data-subject.service'
+import { DomSanitizer } from '@angular/platform-browser'
+
+@Component({
+  selector: 'app-data-export',
+  templateUrl: './data-export.component.html',
+  styleUrls: ['./data-export.component.scss']
+})
+export class DataExportComponent implements OnInit {
+
+  public captchaControl: FormControl = new FormControl('', [Validators.required, Validators.minLength(5)])
+  public formatControl: FormControl = new FormControl('', [Validators.required])
+  public captcha: any
+  private dataRequest: any = undefined
+  public confirmation: any
+  public error: any
+  public lastSuccessfulTry: any
+  public presenceOfCaptcha: boolean = false
+  public userData: any
+
+  constructor (public sanitizer: DomSanitizer, private imageCaptchaService: ImageCaptchaService, private dataSubjectService: DataSubjectService) { }
+  ngOnInit () {
+    this.needCaptcha()
+    this.dataRequest = {}
+  }
+
+  needCaptcha () {
+    let nowTime = new Date()
+    let timeOfCaptcha = localStorage.getItem('lstdtxprt') ? new Date(JSON.parse(String(localStorage.getItem('lstdtxprt')))) : new Date(0)
+    if (nowTime.getTime() - timeOfCaptcha.getTime() < 300000) {
+      this.getNewCaptcha()
+      this.presenceOfCaptcha = true
+    }
+  }
+
+  getNewCaptcha () {
+    this.imageCaptchaService.getCaptcha().subscribe((data: any) => {
+      this.captcha = this.sanitizer.bypassSecurityTrustHtml(data.image)
```

Comment:
### <sub><img src="https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg" width="20"/></sub>&nbsp; Cross-site Scripting (XSS)

Unsanitized input from data from a remote resource flows into bypassSecurityTrustHtml, where it is used to render an HTML page returned to the user. This may result in a Cross-Site Scripting attack (XSS).

<h6>
<a href="https://cwe.mitre.org/data/definitions/79.html">CWE-79</a> | <a href="https://docs.snyk.io/manage-risk/prioritize-issues-for-fixing/priority-score">Priority score 584</a> | Line 46</h6>